test/org/zaproxy/zap/extension/pscanrulesAlpha/CSPNonceScannerUnitTest.java

```diff
@@ -0,0 +1,150 @@
+		msg.setResponseHeader("HTTP/1.1 200 OK\r\n" +
+				      "Content-Security-Policy: default-src: 'none'\r\n");
+
+		// When
+		rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+		// Then
+		assertThat(alertsRaised.size(), is(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfCSPMetaWithoutNonceSrc() throws Exception {
+		// Given
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+		msg.setResponseHeader("HTTP/1.1 200 OK\r\n" +
+				      "Content-Type: text/html\r\n");
+		msg.setResponseBody("<head><meta http-equiv=\"Content-Security-Policy\" content=\"script-src 'self'\"></head>");
+
+		// When
+		rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+		// Then
+		assertThat(alertsRaised.size(), is(0));
+	}
+
+	@Test
+	public void shouldRaiseInvalidNonceAlertIfCSPMetaWithInvalidNonce() throws Exception {
+		// Given
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+		msg.setResponseHeader("HTTP/1.1 200 OK\r\n" +
+				      "Content-Type: text/html\r\n");
+		msg.setResponseBody("<head><meta http-equiv=\"Content-Security-Policy\" content=\"script-src nonce-not!-a-nonce?\"></head>");
+
+		// When
+		rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+		// Then
+		assertThat(alertsRaised.size(), is(1));
+		assertThat(alertsRaised.get(0).getEvidence(), is("not!-a-nonce?"));
+	}
+
+	@Test
+	public void shouldRaiseInvalidNonceAlertIfCSPHeadersWithInvalidNonces() throws Exception {
+		// Given
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+		msg.setResponseHeader("HTTP/1.1 200 OK\r\n" +
+				      "Content-Security-Policy: script-src nonce-not!-a-nonce?\r\n" +
+				      "Content-Security-Policy: script-src nonce-also-not!-a-nonce?\r\n");
+
+		// When
+		rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+
+		// Then
+		assertThat(alertsRaised.size(), is(2));
+		assertThat(alertsRaised.get(0).getEvidence(), is("not!-a-nonce?"));
+		assertThat(alertsRaised.get(1).getEvidence(), is("also-not!-a-nonce?"));
+	}
+
+	@Test
+	public void shouldRaiseNonceReuseAlert() throws Exception {
+		// Given
+		HttpMessage msg = new HttpMessage();
+		msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
+		msg.setResponseHeader("HTTP/1.1 200 OK\r\n" +
+				      "Content-Security-Policy: script-src nonce-reused\r\n");
+
+		HttpMessage msg2 = new HttpMessage();
+		msg2.setRequestHeader("GET https://www.example.com/test/2 HTTP/1.1");
+		msg2.setResponseHeader("HTTP/1.1 200 OK\r\n" +
+				       "Content-Security-Policy: script-src nonce-reused\r\n");
+
+		// When
+		rule.scanHttpResponseReceive(msg, -1, this.createSource(msg));
+		rule.scanHttpResponseReceive(msg2, -1, this.createSource(msg2));
+
+		// Then
+		assertThat(alertsRaised.size(), is(1));
+		assertThat(alertsRaised.get(0).getEvidence(), is("reused"));
```

Comment:
As with the earlier comment, how is this actually going to be checked? The passive scanner only evaluates one message at a time how is `msg2` checked against `msg`?